src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5439,14 +5439,26 @@ private void ParseTypeArgumentList(out SyntaxToken open, SeparatedSyntaxListBuil
             // remaining types & commas
             while (true)
             {
+
                 if (this.CurrentToken.Kind == SyntaxKind.GreaterThanToken)
                 {
                     break;
                 }
-                else if (this.CurrentToken.Kind == SyntaxKind.CommaToken || this.IsPossibleType())
+                else if (this.CurrentToken.Kind == SyntaxKind.CommaToken)
                 {
-                    types.AddSeparator(this.EatToken(SyntaxKind.CommaToken));
-                    types.Add(this.ParseTypeArgument());
+                    eatCommaAndTypeArgument();
+                }
+                else if (this.IsPossibleType())
+                {
+                    var nextToken = this.PeekToken(1);
+                    if (nextToken.Kind == SyntaxKind.GreaterThanToken || nextToken.Kind == SyntaxKind.CommaToken) // presumably missing a comma
```

Comment:
comment is not really useful.  It would be better to have a deeper explanation of what's going on here.